A69-crl-enhancements.md

```diff
@@ -195,32 +194,28 @@ class CrlProvider {
-
-There is already plumbing in these spots for the `crl_directory`, so it shouldn't be too difficult to plumb this around.
-
-Currently, the code that consumes the crl_directory and configures BoringSSL looks like this, in ssl_transport_security.cc:
-
-```C
-      if (options->crl_directory != nullptr &&
-          strcmp(options->crl_directory, "") != 0) {
-        gpr_log(GPR_INFO, "enabling server CRL checking with path %s",
-                options->crl_directory);
-        X509_STORE* cert_store = SSL_CTX_get_cert_store(impl->ssl_contexts[i]);
-        X509_STORE_set_verify_cb(cert_store, verify_cb);
-        if (!X509_STORE_load_locations(cert_store, nullptr,
-                                       options->crl_directory)) {
-          gpr_log(GPR_ERROR, "Failed to load CRL File from directory.");
-        } else {
-          X509_VERIFY_PARAM* param = X509_STORE_get0_param(cert_store);
-          X509_VERIFY_PARAM_set_flags(
-              param, X509_V_FLAG_CRL_CHECK | X509_V_FLAG_CRL_CHECK_ALL);
-          gpr_log(GPR_INFO, "enabled server CRL checking.");
-        }
-      }
-```
+We will add a `grpc::experimental::CrlProvider` that can be configured in `grpc::experimental::TlsChannelCredentialOptions`. This value and `set_crl_directory` cannot both be configured. The Provider will be configured in these options by the user and supplied as a pointer to grpc. Any handshake happening with a provider configured should share this provider, so the user can configure this to be shared across many connections if desired.
+
+These will be passed down from the `TlsCreds` layer to the security connector layer, and from there to the TSI and the ssl_transport_security layer which actually constructs the handshakers and interacts with BoringSSL.
+
+The footprint of the TlsCredentialProvider is more complex than needed for the scope of our CRL work, largely because we are supporting X509 CRL files and not something that needs to make calls such as OSCP. Thus, we can look at the footprint of the custom tls CertificateVerifier to inform how these user-facing options cascade through the code - the provider will have an API that takes certificate information and returns the CRL content. There is already plumbing in these spots for the `crl_directory`, so it shouldn't be too difficult to plumb this around.
```

Comment:
I suggest removing this paragraph, since it doesn't really seem to add any useful information from the perspective of this document.

Reply:
Done